zips/draft-ecc-quantum-recoverability.md

```diff
@@ -0,0 +1,1513 @@
+with $\mathsf{use\_qsk} = \mathsf{true}$ to derive $\mathsf{nk}$, $\mathsf{qsk}$,
+$\mathsf{qk}$, and (using the $\mathsf{ak}$ output by the DKG protocol)
+$\mathsf{rivk\_ext}$, as specified in § 4.2.3 ‘Orchard Key Components’.
+
+```mermaid
+graph BT
+  classDef default stroke:#111111;
+  classDef func fill:#d0ffb8;
+  classDef cefunc fill:#a0e0a0;
+  classDef sensitive fill:#ffb0b0;
+  classDef semi_sensitive fill:#ffb0ff;
+  classDef circuit stroke:#000000, fill:#fffff0;
+  classDef spacer opacity:0;
+
+  sk([sk]):::sensitive --> Hnk[H<sup>nk</sup>]:::func ----> nk([nk])
+  sk --> Hqsk[H<sup>qsk</sup>]:::func --> qsk([qsk]):::sensitive
+  qk ==> Hrivk_ext
+  nk ==> Hrivk_ext
+  FROST_DKG --> ak([ak]) --> FROST_Sign
+  FROST_DKG ~~~ s( ):::spacer ~~~ FROST_Sign
+  FROST_DKG --> ask([ask<sub>i</sub>]):::sensitive --> FROST_Sign
+  ak ==> Hrivk_ext[H<sup>rivk_ext</sup>]:::func
+  Hrivk_ext ==>|use_qsk| rivk_ext([rivk_ext])
+
+  HWW1:::circuit
+  subgraph HWW1[<div style="margin:1.6em;font-size:20px"><br><br><br><br><br><br><br><br><br><b>SoK<sup>qsk</sup></b></div>]
+    qsk([qsk]):::sensitive ==> Hqk[H<sup>qk</sup>]:::cefunc
+    Hqk ==> qk([qk]):::semi_sensitive
+  end
+```
+
+The protocol MUST ensure that all participants obtain the same values for
+$\mathsf{ak}$, $\mathsf{nk}$, and $\mathsf{rivk\_ext}$. (Provided that the
+participants have followed § 4.2.3, this implies that they have the same
+$\mathsf{qsk}$ and $\mathsf{qk}$, by collision resistance of $\mathsf{H^{qk}}$
+and $\mathsf{H^{rivk\_ext}}$.)
+
+Each participant MUST treat $\mathsf{qsk}$ as a secret held at least as
+securely and reliably as $\mathsf{ask}$. Note that $\mathsf{qsk}$ will
+not be needed to sign unless and until the Recovery Protocol is needed,
+but it is essential that it be retained, otherwise funds could be lost if
+and when the current Orchard protocol is disabled.
+
+The Recovery Protocol will still require a RedDSA signature verifiable by
+the Spend validating key $\mathsf{ak}$, in addition to knowledge of
+$\mathsf{qsk}$. Although RedDSA is not secure in the long term against a
+quantum or discrete-log-breaking adversary, the most likely eventuality is
+that attacks against it will be difficult for some years after the current
+Orchard protocol is disabled. During this period, checking this RedDSA
+signature in the Recovery Protocol will ensure that spend authorization by
+a $t$-of-$n$ threshold of participants continues to be needed against
+classical adversaries. This retains the usual advantage of FROST that the
+parties can sign using their shares *without* reconstructing the Spend
+authorizing key $\mathsf{ask}$. Coalitions of fewer than $t$ of the
+participants will be unable to authorize spends as long as they do not
+have access to a sufficiently powerful quantum computer.
+
+Note that a quantum adversary may be able to steal the funds with only
+access to $\mathsf{qsk}$, which is held by every participant. Therefore,
```

Comment:
I am considering changing this (not in this PR) to allow designating a subset of participants that will have $\mathsf{qsk}$. Those participants will give the remaining ones $\mathsf{qk}$, $\mathsf{ak}$, and $\mathsf{nk}$.

(It still follows that if all the participants have the same $\mathsf{ak}$, $\mathsf{nk}$, and $\mathsf{rivk\_ext}$, then they all have the same $\mathsf{qk}$. But the participants who didn't have $\mathsf{sk}$ can no longer check directly that the ones that did, have $\mathsf{qsk}$. It may be desirable for each participant that is supposed to have $\mathsf{qsk}$ to provide a proof of knowledge that they have it.)